### ISSUE TITLE
Issue with hiera lookup and boolean value

### ISSUE THREAD

**Issue by Vincent--** [ajcrowe/puppet-supervisord#86]
I'm currently using hiera to create my program:

```
foo_autostart: true

supervisor_programs:
    'foo':
        command        : "foo"
        [...]
        autostart      : "%{hiera('foo_autostart')}"
```

And for some reason, I have to use a hiera lookup for the autostart value.
With puppet 3.8.4 and hiera 1.3.4 and the code above, this ends up with `autostart = "true"`. As the module is checking if the value is a real boolean, the compilation fails because "true" is a string.

Based on my knowledge, it's not possible to do a hiera lookup with boolean.

So I'm wondering if we cannot replaced this check:
```
if $autostart { validate_bool($autostart) }
```

by this one (already used for the autorestart param):
```
if $autostart { if !is_bool($autostart) { validate_re($autostart, ['true', 'false']) } }
```

or even if we cannot use the str2bool func like this (not tested, but you get the idea):
```
if $autostart { if !is_bool($autostart) { $autostart=str2bool($autostart) } }
```
See https://forge.puppetlabs.com/puppetlabs/stdlib#str2bool

I'm happy to do the PR, but need to know which solution you want to be implemented...

**Comment by ajcrowe** [comment]
This is strange, are you sure that doing `"%{hiera('foo_autostart')}"` isn't just added the quotes?

We use json for our hiera backend and haven't seen any problems using booleans for autostart.

**Comment by Vincent--** [comment]
You are right, doing `"%{hiera('foo_autostart')}"` adds the quotes but when using the yaml format, you have to use the quote for lookup, see:
https://docs.puppet.com/hiera/3.1/variables.html#in-data

> Note: With YAML data sources, interpolating into hash keys means those hash keys must be quoted.

I've already tried to remove the quote but then the catalog compilation fails...

**Comment by ajcrowe** [comment]
Is there a reason you would want to do a lookup for this? Just taking this to it's logical conclusion, wouldn't this mean all booleans would need this change to avoid similar problems?

**Comment by Vincent--** [comment]
My use case is not very simple but basically, I want to be able to override the autostart value.
As I don't use deep merging, with this lookup, I just need to write `foo_autostart: false` in another hiera file which has bigger priority in the hierarchy.
If I don't use the lookup, I have to override the full `supervisor_programs:` hash, so that mean the code is completely redundant and hard to maintain.

TL;DR; I reckon all the booleans might need this change too. But on the other hand, I believe the hiera lookup works fine for boolean in puppet 4.x (not sure of that however)

**Comment by ajcrowe** [comment]
I've merged #91 to solve this as there are very few only boolean values.
